src/themes/OLH/templates/journal/search.html

```diff
@@ -24,10 +33,35 @@ <h3>{% trans "Articles" %}</h3>
                     <form method="POST">
                         {% csrf_token %}
                         <div class="section">
+                             <label>{% trans "Search" %}
+                                <input type="text" name="article_search" value=""/>
+                            </label>
+                        </div>
+                        <div class="section">
+                            <label>{% trans "Keywords" %}</label>
+
+                                {% for keyword in all_keywords %}
+                                    <a href="{% url 'search' %}?keyword={{keyword.word}}">{{ keyword.word }}</a>{% if not forloop.last %}, {% endif %}
+                                {% endfor %}
 
                         </div>
+                        <div class="section">
+                            <label>{% trans "Sort" %}
+                                <select name="sort">
+                                    <option value="-date_published" {% if sort == "date_published" %} selected="selected" {% endif %}>
+                                        {% trans "Date" %}
+                                    </option>
+                                    <option value="title" {% if sort == "title" %} selected="selected" {% endif %}>
+                                        {% trans "Title" %}
+                                    </option>
+                                </select>
+                            </label>
+                        </div>
                         <div class="section">
                             <button type="submit" class="button">{% trans "Filter" %}</button>
+                             {% if search_filters %}
```

Comment:
Search filters is an invalid reference for the context of the view that renders this template.
(this applies to changes on all three themes)